src/main/java/org/terasology/launcher/config/Config.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.config;
+
+import org.terasology.launcher.packages.Package;
+
+import java.nio.file.Path;
+import java.util.Locale;
+import java.util.Objects;
+
+/**
+ * Stores all application-wide configuration values.
+ * It is immutable and can be instantiated only with
+ * the help of a {@link Builder} instance.
+ */
+public final class Config {
+    private final GameConfig gameConfig;
+    private final Locale locale;
+    private final transient Path launcherDir;
+    private final boolean checkUpdatesOnLaunch;
+    private final boolean closeAfterGameStarts;
+    private final boolean cacheGamePackages;
+    private final Package selectedPackage;
+
+    private Config(final Builder builder) {
+        this.gameConfig = builder.gameConfig;
+        this.locale = builder.locale;
+        this.launcherDir = builder.launcherDir;
+        this.checkUpdatesOnLaunch = builder.checkUpdatesOnLaunch;
+        this.closeAfterGameStarts = builder.closeAfterGameStarts;
+        this.cacheGamePackages = builder.cacheGamePackages;
+        this.selectedPackage = builder.selectedPackage;
+    }
+
+    public GameConfig getGameConfig() {
+        return gameConfig;
+    }
+
+    public Locale getLocale() {
+        return locale;
+    }
+
+    public Path getLauncherDir() {
+        return launcherDir;
+    }
+
+    public boolean isCheckUpdatesOnLaunch() {
+        return checkUpdatesOnLaunch;
+    }
+
+    public boolean isCloseAfterGameStarts() {
+        return closeAfterGameStarts;
+    }
+
+    public boolean isCacheGamePackages() {
+        return cacheGamePackages;
+    }
+
+    public Package getSelectedPackage() {
+        return selectedPackage;
+    }
+
+    /**
+     * Provides a pre-filled {@link Builder} instance
+     * with all configurations copied from this. Use it
+     * to create a new {@link Config} instance with
+     * modified configurations.
+     *
+     * @return a pre-filled {@link Builder} instance
+     */
+    public Builder rebuilder() {
```

Comment:
I'm not sure about this method - I would rather make it a variant of instantiating a new builder:
```java
Config.Builder builder = Config.builder();
...
Config obj = builder.build();

Config.Builder rebuilder = Config.builder(obj);
```

I'm not even sure about the static `builder` method but we can keep it for now.